conans/test/unittests/tools/files_patch_test.py

```diff
@@ -278,3 +278,41 @@ def _build_and_check(self, tmp_dir, file_path, text_file, msg):
 
         content = load(text_file)
         self.assertEqual(content, msg)
+
+    def test_fuzzy_patch(self):
+        conanfile = dedent("""
+            from conans import ConanFile, tools
+            import os
+
+            class ConanFileToolsTest(ConanFile):
+                name = "fuzz"
+                version = "0.1.0"
+                exports_sources = "*"
+
+                def build(self):
+                    tools.patch(patch_file="fuzzy.patch", fuzz=True)
```

Comment:
IMO, what we should test here in Conan (see https://github.com/conan-io/conan-center-index/issues/214 which was the issue motivating these changes) is that **by default this function fails if the patch is fuzzy**.

So 

```suggestion
                    tools.patch(patch_file="fuzzy.patch")
```

and it should raise a `ConanException` saying something about a fuzzy patch.